src/target/riscv/riscv-013.c

```diff
@@ -195,6 +195,12 @@ typedef struct {
 	bool abstract_read_fpr_supported;
 	bool abstract_write_fpr_supported;
 
+	/* Track whether the hart is currently halted. This is more precise than
+	 * target->state, because it'll reflect halted status even if we're
+	 * pretending to the client that the hart is running (ie. during
+	 * examine()). */
+	bool halted[RISCV_MAX_HARTS];
```

Comment:
should you call this something like "really_halted" to distinguish?

Reply:
Yes, that's better.